src/routes/api/auth/index.ts

```diff
@@ -43,6 +42,47 @@ const plugin: FastifyPluginAsyncTypebox = async (fastify) => {
       return { message: 'Invalid username or password.' }
     }
   )
-}
+  fastify.post(
+    '/update',
+    {
+      schema: {
+        body: UpdateCredentialsSchema,
+        response: {
+          200: Type.Object({
+            message: Type.String()
+          }),
+          401: Type.Object({
+            message: Type.String()
+          })
+        },
+        tags: ['Authentication']
+      }
+    },
+    async function (request, reply) {
+      const { username } = request.body
+      const user = await fastify.repository.find<Auth>('users', {
+        select: 'username, password',
+        where: { username }
+      })
+
+      if (user) {
+        const isPasswordValid = await fastify.compare(request.body.password, user.password)
 
+        if (isPasswordValid) {
+          const hashedPassword = await fastify.hash(request.body.newPassword || request.body.password)
+          await fastify.repository.update('users', {
+            data: { password: hashedPassword, username: request.body.newUsername || request.body.username },
```

Comment:
I don't see it is a good idea to update `username` without database unique constraint or transaction checking on data.